File: src/main/java/kr/mjc/rachel/spring/jdbc/AppTest.java
package kr.mjc.rachel.spring.jdbc;

import lombok.extern.slf4j.Slf4j;
import org.slf4j.Logger;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

import java.util.function.BiConsumer;

@Slf4j
public class AppTest {
  public <T> void test(Class<T> clazz, BiConsumer<T, Logger> biConsumer) {
    ApplicationContext applicationContext =
        new AnnotationConfigApplicationContext(AppConfig.class);

    T dao = applicationContext.getBean(clazz);
    biConsumer.accept(dao, log);
  }
}

File: src/main/kotlin/kr/mjc/jacob/spring/jdbc/AppTest.kt
package kr.mjc.jacob.spring.jdbc

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.AnnotationConfigApplicationContext

class AppTest {
  fun <T> test(clazz: Class<T>, accept: (T, Logger) -> Unit) {
    val log = LoggerFactory.getLogger(object {}.javaClass.enclosingClass)
    val context = AnnotationConfigApplicationContext(AppConfig::class.java)
    val dao: T = context.getBean(clazz)
    accept(dao, log)
  }
}